app/Model/Behavior/ProvisionerBehavior.php

```diff
@@ -29,6 +29,11 @@
 App::uses('CakeSession', 'Model/Datasource');
 
 class ProvisionerBehavior extends ModelBehavior {
+  
+  // Auxiliary property that faciliates manual provisioning
+  // Set once per request
+  private $isRequestToProvision = false;
```

Comment:
Since this is just a flag passed between requestToProvision and manualProvision, why not just pass it as a flag?

Also (and I admit this is somewhat arbitrary) it seems like requestToProvision should be the "core" function and manualProvision should be the wrapper (in which case the flag can simply be a `$manual=false` parameter added to requestToprovision).

Reply:
I renamed the `manualProvision` to` requestToProvision` and made the `$manual` a flag